### PR TITLE
remove force-multi and fix None val check in base tuner in autotuning

### PR DESCRIPTION
**Pull request by cli99**
This PR fixes the following bugs in autotuning:
1. the autotuning scheduler does not add force-multi to the experiment script and uses what users pass in
2. the initial best metric value could be None in the tuning and needs to be checked

### CODE REVIEW

File: deepspeed/autotuning/scheduler.py [hunk 1]
```diff
@@ -333,7 +333,6 @@ def run_experiment(exp: dict, reservations, user_script, user_args):
     include_str = include_str[:-1]
     master_port = exp["master_port"]
     exp["launcher_args"] = [
-        "--force_multi",
```
Comment (tjruwase) [inline deepspeed/autotuning/scheduler.py:336]: What is the downside of this option?

Comment (jeffra) [reply deepspeed/autotuning/scheduler.py:336]: @cli99 and I were talking about this before the holidays. If I recall correctly, she was seeing an issue where if password-less ssh isn't setup correctly than `--force_multi` and autotuning in general will fail. Even on our ITP nodes we saw an issue where if the destination host was set to `localhost` this didn't match the ssh alias (i.e., worker-N) and hung with a password prompt. We do need `--force_multi` if we're doing auto-tuning across multiple nodes though so not sure what the best default is here? Can auto-tuning detect if we're trying to tune across multiple hosts and use `--force_multi` in that case only?

Comment (cli99) [reply deepspeed/autotuning/scheduler.py:336]: We ask the users to specify --force_multi in the training script if it's needed. The autotuner does not add it explicitly but passes what the train script uses.

Comment (jeffra) [reply deepspeed/autotuning/scheduler.py:336]: If a single candidate job requires > 1 node then yes we don't need `--force_multi`. However, if a single candidate job requires only 1 node and we have 2+ nodes available in the scheduling pool then we will require `--force_multi` for it to launch the candidate job on a remote node. We can sync offline, @tjruwase and I are discussing this now as well :)

Comment (cli99) [reply deepspeed/autotuning/scheduler.py:336]: In that case, due to line https://github.com/microsoft/DeepSpeed/blob/master/deepspeed/launcher/runner.py#L350, multi_node_exec is also set. We don't need --force_multi explicitly?

**Comment by rocm-mici** [comment]
Can one of the admins verify this patch?

**Comment by loadams** [comment]
These changes were already applied in other PRs, closing this one.

**Pull request closed without merging**